core/dataset.cpp

```diff
@@ -252,6 +266,19 @@ void Dataset::setAttr(const std::string &attrName, Variable attr) {
   m_attrs.insert_or_assign(attrName, std::move(attr));
 }
 
+/// Set (insert or replace) the masks for the given label name.
+///
+/// Note that the label name has no relation to names of data items.
+void Dataset::setMasks(const std::string &labelName, Variable masks) {
+  setDims(masks.dims());
+
+  if (masks.dtype() != DType::Bool) {
+    throw std::logic_error("Masks can only be of boolean type.");
```

Comment:
If we do that then the `Variable::operator|` needs to account for the cases that aren't `Bool`/`bool`/`int`. `operator|` isn't available for float/double

Reply:
It will through. I think this is fine. Later we can consider supporting those cases, if required. For now, I would just keep the containers generic, provided that there is no downside. Failure of certain operations is not a real downside in my opinion, since you could make the same argument for data that is strings, which cannot be added (we would not forbid string data).

Reply:
I see, so I would remove the check here and add a explicit `bool` lambda overload to the transform, which does the OR, and an auto that throws (for other dtypes)?

Reply:
There are no changes required to the transform: It only supports the types (type pairs) you list explicitly and will throw for others, so it should already have the desired effect.